### PR TITLE
Create new filter to custom log messages

### PR DESCRIPTION
**Pull request by davefx**
I've created a new filter for log messages, so I can add extra info to the log messages written by WooCommerce, its extensions or related plugins.

In my project, I'm using this to prepend a checksum to the message that associates every message with the request that created it.

### CODE REVIEW

**Comment by claudiosmweb** [comment]
@mikejolley thoughts?

I just only don't like the filter name, we can change to:

```php
$message = apply_filters( 'woocommerce_logger_add_message', $message, $handle );
```

**Comment by davefx** [comment]
OK. Filter name changed

File: includes/class-wc-logger.php [hunk 1]
```diff
@@ -98,6 +98,7 @@ public function add( $handle, $message ) {
 		$result = false;
 
 		if ( $this->open( $handle ) && is_resource( $this->_handles[ $handle ] ) ) {
+		    $message = apply_filters('woocommerce_logger_add_message', $message, $handle);
```
Comment (mikejolley) [inline includes/class-wc-logger.php]: Needs a space between the parenthesis.

**Review by mikejolley — CHANGES_REQUESTED** [review]

**Comment by davefx** [comment]
Done

File: includes/class-wc-logger.php (hunk 1 above)
Comment (davefx) [inline includes/class-wc-logger.php]: Done

Comment (davefx) [inline includes/class-wc-logger.php]: Fixed in the pull request, @mikejolley

**Pull request merged**